### PR TITLE
FM-1 Add favourite wfs

### PR DESCRIPTION
**Pull request by TomasHodor**
Signed-off-by: thodor <thodor@frinx.io>

### CODE REVIEW

File: src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js [hunk 1]
```diff
@@ -93,11 +94,13 @@ class WorkflowDefs extends Component {
...
     updateFavourite(data) {
+        this.setState({
+            loading: true
+        });
         data.description = data.description.includes(", FAVOURITE")
             ? data.description.replace(", FAVOURITE","")
             : data.description += ", FAVOURITE";
-        http.put('/api/conductor/metadata/', [data]).then(res => {
-        })
+        http.put('/api/conductor/metadata/', [data]).then(() => window.location.reload());
```
Comment (ChmaraX) [inline src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js]: window.location.reload is anti-pattern, updating data cant be solved with reloading window

File: src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js [hunk 2]
```diff
@@ -93,11 +94,13 @@ class WorkflowDefs extends Component {
     }
 
     updateFavourite(data) {
+        this.setState({
+            loading: true
```
Comment (ChmaraX) [inline src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js]: "loading" is set to true but never get set back to "false", remounting component with window.location.reload and reseting state can cause unexpected behavior

File: src/api/conductor.js [hunk 3]
```diff
@@ -42,7 +42,7 @@ router.put('/metadata', async (req, res, next) => {
     try {
         let workflowDesc = req.body;
```
Comment (ChmaraX) [inline src/api/conductor.js]: workflowDesc variable is redundant

File: src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js [hunk 4]
```diff
@@ -198,6 +201,9 @@ class WorkflowDefs extends Component {
                      </Form.Group>
                  </Col>
              </Row>
+             <div className={this.state.loading?"loading":""}>
+                 <i className={this.state.loading?"fa fa-spinner fa-pulse fa-5x fa-fw":""}/>
+             </div>
```
Comment (ChmaraX) [inline src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js]: loading icon in WorkflowDefs is not necessary at all, you dont need to see how long does saving favourites takes or if its finished saving

File: src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js (hunk 2 above)
Comment (TomasHodor) [reply src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js]: Done

File: src/api/conductor.js (hunk 3 above)
Comment (TomasHodor) [reply src/api/conductor.js]: Done

File: src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js (hunk 1 above)
Comment (TomasHodor) [reply src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js]: Done

File: src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js (hunk 4 above)
Comment (TomasHodor) [reply src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js]: Done

File: src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js [hunk 5]
```diff
@@ -94,13 +93,11 @@ class WorkflowDefs extends Component {
...
     updateFavourite(data) {
-        this.setState({
-            loading: true
-        });
         data.description = data.description.includes(", FAVOURITE")
             ? data.description.replace(", FAVOURITE","")
             : data.description += ", FAVOURITE";
-        http.put('/api/conductor/metadata/', [data]).then(() => window.location.reload());
+        http.put('/api/conductor/metadata/', [data]).then(res => {
+        });
```
Comment (ChmaraX) [inline src/components/workflows/workflowList/WorkflowDefs/WorkflowDefs.js]: please remove ".then(res => { });" if it does nothing.

**Pull request merged**